### ISSUE TITLE
Test the tN tool with the Hindi tN

### ISSUE THREAD

**Issue by benjore** [unfoldingWord/translationCore#6050]
### User Story

As a [type of user], I want [some goal] so that [some reason].

### Features / Specifications

- [ ] Before releasing the tN tool, check it using the Hindi tN to make sure it works as expected. 
- [ ] https://git.door43.org/Door43-Catalog/hi_tn
- [ ]

### Definition of Done

- [ ]
- [ ]
- [ ]

**Comment by elsylambert** [comment]
#1.2.0 (4b9a560)
Got an alert when I click on the Menu options. When clicked Ok on the alert, it shows a broken link on tN card but does not show which verse is broken.
![image (1)](https://user-images.githubusercontent.com/51806180/60215628-0caf9400-9836-11e9-9728-5f1f3887863c.png)
[hi_els_phm_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3331948/hi_els_phm_book.zip)


[tN Alert message.zip](https://github.com/unfoldingWord-dev/translationCore/files/3331950/tN.Alert.message.zip)

**Comment by elsylambert** [comment]
This alert message comes up when the GL is selected as English as well. It does not shows the broken link on the verse but shows the green check even when the selection is lost.

**Comment by PhotoNomad0** [comment]
@elsylambert - It looks like in tN, when you click save/continue or navigate to invalid verse it is actually warning about previous invalidations, not new invalidations.

PS - I really like the videos with mouse highlighted and button clicks! @cckozie

**Comment by PhotoNomad0** [comment]
**Notes:**
- it looks like the selections are saved with single white space between words where the target translation may have many - this is causing validation trouble when selections are compared with original because they are not found.
- this seems to be due to issue that verses are rendered in verseCheck and Scripture Pane with extra spaces removed.

Example from phm: 
` तेरी   इच्छा  चाहा  बिना  कुछ न  चाहा  कि  तेरा     यह  उपकार आनन्द  दबाव  से    नहीं     पर  आनन्द से हो `
 (lots of extra spaces that are not obvious with proportional fonts).

**Comment by PhotoNomad0** [comment]
@benjore - I'm also seeing problems with extra spaces in bn projects.  I am wondering if all the computer generated translations have this feature.

**Comment by PhotoNomad0** [comment]
@benjore - you probably need bring this up with the content team.  There seems to be an issue with the computer generated translations that have extra spaces between words (have seen as many 4 between words).  When we save the selections we have been normalizing to have only a single space between words.  This could cause trouble with the scripts the content team uses to generate tW and tN resources for the gateway language.
- We can change the selections to have exact match of spacing between words, but that means that if an editor cleans up the text with single spacing it will invalidate many selections. (direct edits or overwrite merge)
- Or we could keep selections to single spacing between words and just match token for token when checking for invalidations, which would give the most flexibility.  But content teams scripts will have to be updated to be more flexible.  May also impact performance of selection invalidation checking.

**Comment by russp41** [comment]
@PhotoNomad0 @benjore I've mentioned this to the Hindi team, but I can't find an example of extra spaces to show them. Where are you seeing those in Philemon? I'm not seeing any in the IRV text or tN at [https://git.door43.org/Door43-Catalog/hi_irv/raw/branch/master/58-PHM.usfm](https://git.door43.org/Door43-Catalog/hi_irv/raw/branch/master/58-PHM.usfm) or [https://git.door43.org/Door43-Catalog/hi_tn/src/branch/master/hi_tn_58-PHM.tsv](https://git.door43.org/Door43-Catalog/hi_tn/src/branch/master/hi_tn_58-PHM.tsv)

**Comment by PhotoNomad0** [comment]
@russp41 - Hmm...interesting.  Here is the PHM text with the alignments parsed out by tC. vs 11:

```
पहले   तेरे कुछ काम काम न था   पर  अब  तेरे  और  दोनों मेरे दोनों के   बड़े काम का है 
```

here is the initial USFM for vs 11.  It is in the old format, but looks good to my eye:

```
\v 11
\zaln-s | x-strong="G35880" x-lemma="ὁ" x-morph="Gr,RD,,,,AMS," x-occurrence="1" x-occurrences="1" x-content="τόν"
\zaln-s | x-strong="G42180" x-lemma="ποτέ" x-morph="Gr,D,,,,,,,,," x-occurrence="1" x-occurrences="1" x-content="ποτέ"
\w पहले|x-occurrence="1" x-occurrences="1"\w*
\zaln-e\*
\zaln-e\*
\zaln-s | x-strong="G47710" x-lemma="σύ" x-morph="Gr,RP,,,2D,S," x-occurrence="1" x-occurrences="1" x-content="σοι"
\zaln-s | x-strong="G08900" x-lemma="ἄχρηστος" x-morph="Gr,NS,,,,AMS," x-occurrence="1" x-occurrences="1" x-content="ἄχρηστον"
\w तेरे|x-occurrence="1" x-occurrences="2"\w*
\w कुछ|x-occurrence="1" x-occurrences="1"\w*
\w काम|x-occurrence="1" x-occurrences="3"\w*
\w काम|x-occurrence="2" x-occurrences="3"\w*
\w न|x-occurrence="1" x-occurrences="1"\w*
\w था|x-occurrence="1" x-occurrences="1"\w*
\zaln-e\*
\zaln-e\*
```

At first glance it seems that the USFM parser has a problem with backward compatibility with our old broken alignments format, and is adding an extra space after each `\zaln-e\*` tag.  In the USFM spec the newline is used in place of a space in the text.

**Comment by PhotoNomad0** [comment]
@russp41  - Found a simple fix for USFM file before import!  I was looking in the wrong place - the content and not the header.  In the USFM file, just remove the line `\usfm 3.0` and it will import correctly without extra spaces.  Unfortunately we used USFM level 3 to determine if the USFM was exported by a new version of tCore with corrected USFM.  

@benjore  - I believe we have an issue to change when we add the USFM level 3 on export.  We should probably add to that issue to change usfm-js to use text inspection to determine whether the USFM is old or new format.

**Comment by PhotoNomad0** [comment]
Making a change to usfm-js to use text inspection to determine whether the alignment USFM is the old or new format.

**Comment by PhotoNomad0** [comment]
@elsylambert - the part we are fixing is usfm import bug that causes extra spacing.  This should probably be in QA watch until the hi content gets reimported.

**Comment by PhotoNomad0** [comment]
Verified USFM import fix in translationCore 1.2.0 (c1dae4b) 

@elsylambert @benjore @cckozie - the full fix for this will take some time since one problem is a content issue on Door43 in the hi bibles (extra spaces), which was created by another content issue in the USFM created by BCS computer generated content (they put in `\usfm 3.0` in the header, but were using the old alignment format that was not usfm 3 compatible).  The latest build will fix the import problem because it does not look at the `\usfm` marker, but instead looks at the content to determine if it is in the old format.

The fixing of the content on Door43 will involve manual work since the projects will have to be reimported.  Since the next release is not going to be soon, for now they could just remove `\usfm 3.0` before they do the import.  This part should probably be in QA watch to check again after the content is updated.

@elsylambert - to test the import, here is one of the BCS source files you can import from local files:  
[58-PHM-hi.usfm.txt](https://github.com/unfoldingWord-dev/translationCore/files/3383567/58-PHM-hi.usfm.txt)

Then launch the WA tool - open scripture pane and edit 1:1 - should not see extra spaces between word 1 & 2 (or any other words). The previous tCore would see the `\usfm 3.0` marker in header and try to process as valid USFM converting all the newlines to spaces.  This resulted a space at the start of each verse and extra spaces wherever the greek words were merged.

Next lanch the tN tool and should not see all the invalid selections alerts when navigating between checks.

**Comment by elsylambert** [comment]
1.2.0 (736956e), Verified with the BCS source files Project. Cannot find any issue with that project in tN. In wA, noticed that there were no extra spaces in the verses.

needs regression.

**Comment by cckozie** [comment]
Relying on Elsy on this one.

**Comment by cckozie** [comment]
1.2.0 (9f6a878)
The occurrence quote is being displayed in English instead of the GL selected on the tN tool card.

**Comment by PhotoNomad0** [comment]
@cckozie - Hmm... it was looking fine here with 58-PHM-hi.usfm.txt example above in the latest develop build.  Then I updated the en and hi resources and see the problem you mentioned.

**Comment by PhotoNomad0** [comment]
@cckozie - Nuts, I cleared out the resources and started again.  Now everything it working.  Can you still reproduce this problem?

**Comment by PhotoNomad0** [comment]
Note: ~looks like the problem here is in the latest Hindi ulb is missing the alignments to OrigLang.~ ~Looks like the GLQuotes in hi tN are in English.~  Looks like the hi alignments are in the irv which is not always opened the GL is changed.

**Comment by PhotoNomad0** [comment]
Verified in translationCore 2.0.0 (acc38de) - selection quote is now shown in hi when hi GL selected.  Also not seeing the missing quote message.

**Comment by cckozie** [comment]
2.0.0 (00bca0b)
I am now seeing the occurrence quote now always in the GL as expected.
English is still being displayed in the smurf bar. See issue #6275
@Elsy - I see that we are now displaying the tN sub-categories in the GL. When you test this, please ensure that the same categories appear on the tN tool card (and are properly sorted alphabetically) when the GL is both English and Hindi.
BTW: I re-failed #6174 because it is not properly filtering the tN's available by language/book.

**Comment by elsylambert** [comment]
Looks good in 2.0.0 (9016823). 
smurf bar issue addressed in- #6275 
Sub-categories in GL- alphabetical order- #6056 
Needs regression.
